docs/apidocs/fake_provider.rst

```diff
@@ -0,0 +1,4 @@
+.. automodule:: qiskit_ibm_runtime.fake_provider
```

Comment:
@jyu00 can you please confirm that you want this module to be publicly exposed in the docs?

Reply:
Kevin confirmed over Slack it's a public API.